testdata/validate-data/generic/content_violation.xml

```diff
@@ -121,4 +121,45 @@
         </uri-prop>
     </resource>
 
+    <!-- Date with start later than end date -->
+
+    <resource label="wrong_date_range_full_JULIAN" restype=":ClassWithEverything" id="wrong_date_range_full_JULIAN">
+        <date-prop name=":testSubDate1">
+            <date>JULIAN:CE:2000-01-01:CE:1900-01-01</date>
```

Comment:
You only have one very simple error category: the start year is bigger than the end year.

I would add some cases where the year is equal, and only the months are wrong, and some where only the days are wrong.